Tribler/community/tunnel/tunnel_community.py

```diff
@@ -405,16 +403,10 @@ def _generate_circuit_id(self, neighbour=None):
 
     def do_circuits(self):
         for circuit_length, num_circuits in self.circuits_needed.items():
-            num_to_build = num_circuits - sum([1 for c in self.circuits.itervalues() if c.goal_hops == circuit_length])
-            self._logger.debug("want %d circuits of length %d", num_to_build, circuit_length)
-
+            num_to_build = num_circuits - sum([1 for _ in self.data_circuits(circuit_length).itervalues()])
+            self._logger.debug("want %d data circuits of length %d", num_to_build, circuit_length)
             for _ in range(num_to_build):
-                try:
-                    if not self.create_circuit(circuit_length):
-                        break
-                except:
-                    self._logger.exception("Error creating circuit while running do_circuits")
-
+                self.create_circuit(circuit_length)
```

Comment:
Let's check the return value of create_circuit, if its false there's no need to continue.